app/models/lesson.server.ts

```diff
@@ -19,3 +21,32 @@ export async function getLesson(slug: string) {
     .then((res) => res.data.data);
   return lesson;
 }
+
+export async function setCompleted(
+  lessonId: string,
+  request: any,
+  markCompleted = true
+) {
+  const token = await currentToken({ request });
+
+  let endpoint = "";
+  if (markCompleted) {
+    endpoint = `${process.env.API_HOST}/lessons/${lessonId}/completed`;
+  } else {
+    endpoint = `${process.env.API_HOST}/lessons/${lessonId}/uncompleted`;
+  }
```

Comment:
Pequena sugestão aqui. Pode ignorar se quiser 😅

```suggestion
  let endpoint = `${process.env.API_HOST}/lessons/${lessonId}/`;
  if (markCompleted) {
    endpoint += "completed";
  } else {
    endpoint += "uncompleted";
  }
```